backend/TreeOfAKind.Domain/UserProfiles/UserNameMustBeUniqueRule.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using TreeOfAKind.Domain.SeedWork;
+
+namespace TreeOfAKind.Domain.UserProfiles
+{
+    public class UserNameMustBeUniqueRule : IBusinessRule
+    {
+        private readonly string _username;
+        private readonly IUsernameUniquenessChecker _uniquenessChecker;
```

Comment:
Actually is there a user name in our domain? It's rather unnecessary since we log in with email.